### DISCUSSION TITLE
Can I modify existing diagrams?

### DISCUSSION THREAD

**Discussion by sammerlahn** [MerrimanInd/drawpyo#49] [Q&A]
I would like to modify previously created diagrams iteratively via python code. Is it possible with DrawPy?
I.e. I need to 
- open an existing file
- find the page I created before (or just the first one)
- find previously created objects (by id or by some internal data on the object)
- add new objects if not existing
- add new objects to new or existing parent containers

How would I be able to that?

**Comment by MerrimanInd** [comment]
Hey @sammerlahn sorry for the long delay on this answer!

That's something we would like to do but it's not yet supported. Drawpyo basically supports a small subset of everything possible in Draw.io but that's fine since it's a one-way sync. Being able to import any Draw.io file, work with even a subset of the functionality in drawpyo, and re-export it is going to be some work to implement. So maybe someday but not yet today.
